README-dev.md

```diff
@@ -0,0 +1,266 @@
+
+## Startup
+
+### Install Tools
+
+Install the following tools:
+* git
+* Steel Bank Common Lisp (SBCL)
+* asdf (comes with SBCL)
+
+You should be able to execute the following programs:
+* `git` (or whatever tool you prefer)
+* `sbcl` Common Lisp REPL
+
+The rest of this document assumes the command-line use of `git`.
+You may prefer a different tool (including VSCode).
+
+#### Check ASDF Version
+
+Some older SBCL distributions may have an old version of ASDF.
+You must have version 3.2 or 3.3.
+To check this start the SBCL REPL and execute:
+```
+*features*
+(:quicklisp :asdf3.3 :asdf3.2 :asdf3.1 :asdf3 :asdf2 :asdf :os-unix
+ :non-base-chars-exist-p :asdf-unicode :arena-allocator :x86-64 :gencgc :64-bit
+ :ansi-cl :common-lisp :elf :ieee-floating-point :linux :little-endian
+ :package-local-nicknames :sb-ldb :sb-package-locks :sb-thread :sb-unicode
+ :sbcl :unix)
+```
+Note the symbols starting with `:asdf`.
+In the above case, in `SBCL 2.3.2` as shown when the REPL starts,
+the highest version of ASDF is `:asdf3.3`.
+This is sufficient for work on Alive-lsp.
+
+### Install Alive
+
+Alive must be installed into VSCode to do testing of Alive-lsp.
+
+Most Alive-lsp development will require working on both Alive and Alive-lsp
+since the two communicate frequently.
+Download the Alive codebase as described in the Alive
+[development notes](https://github.com/nobody-famous/alive/blob/master/README-dev.md).
+
+### Acquiring the Code
+
+Alive-lsp development is done on a fork of the repository.
+Pull Requests (PRs) are made from a branch in that fork
+into the Alive repository.
+
+* [Create a fork](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+  of the Alive=lsp code into your own `github` account.
+* [Clone your fork](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+  onto your development machine.
+
+This is the same process [documented](https://github.com/nobody-famous/alive/blob/master/README-dev.md)
+for Alive development.
+
+### Inital Steps
+
+Launch VSCode and open a workspace with the downloaded Alive-lsp directory.
+Edit the `.vscode/settings.json` file to set `alive.lsp.install.path`
+to point to the root directory where the Alive-lsp code is installed.
+For example:
+```
+    "alive.lsp.install.path": "/home/<user>/work/lisp/alive-lsp",
+```
+This setting will cause the Alive extension installed in VSCode to find the
+source code for the Alive-lsp project instead of downloading the latest release.
+That way edits made to Alive-lsp code will be executed during testing.
+_Avoid checking this change back into the repository_.
+
+Show the `Output` view at the bottom of the screen (`<ctrl-K><ctrl-H>` or
+the **Output: Focus on Output View** command) or just select the `Output` tab on the bottom panel.
+On the right part of the title bar there is a dropdown to choose output from different threads.
+Choose the `Alive LSP` thread which should be available when
```

Comment:
The "Alive LSP" window shows the output of the SBCL process.

The "Alive Client" window shows the protocol trace between the server and client.

Reply:
I'm seeing "Alive Log" instead of "Alive Client". I rearranged the doc using this string.

Reply:
OK, I'm seeing "Alive Client". I moved the discussion of the three Alive output views to the **Debugging** section and fixed this part of the doc to gloss over the choice for now and to reference that section.